### PR TITLE
jsHomeWorkWeek1-Mahmoud

### PR DESCRIPTION
**Pull request by mahmoudbd**
jsHomeWorkWeek1-Mahmoud

### CODE REVIEW

**Comment by mahmoudbd** [comment]
jsHomeWorkW1

File: Week1/Homework/js-exercises/Exercise1.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+console.log('Hello world!'); // English
+console.log('مرحبا بالعالم'); // Arabic
+console.log('Hallo Wereld!'); // Dutch
+console.log('こんにちは世界!'); // Japans
+console.log('Здравей свят!'); // Bulgaars
+console.log('Bonjour le monde!'); // Frans
+console.log('Selam Dünya!'); // Turks
+console.log('Witaj świecie!'); // Pools
+console.log('Hai dunia!'); // Maleis
+console.log('Hola Mundo!'); // Spaans
```
Comment (eidosam) [inline Week1/Homework/js-exercises/Exercise1.js:11]: 💯

File: Week1/Homework/js-exercises/Exercise2.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+'use strict';
+console.log("I'm awesome");
+console.log('I/m awesome');
```
Comment (eidosam) [inline Week1/Homework/js-exercises/Exercise2.js]: This solves the problem with the error `SyntaxError` but doesn't show the requested output.

Maybe this document can help to solve this https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String#Escape_notation

File: Week1/Homework/js-exercises/Exercise3.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+'use strict';
+let numberX;
+console.log('The value of X is a number');
```
Comment (eidosam) [inline Week1/Homework/js-exercises/Exercise3.js]: After you ran/executed this script; did it meet your expectation? I mean, is `numberX` is really a number at this point?

File: Week1/Homework/js-exercises/Exercise3.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+'use strict';
+let numberX;
+console.log('The value of X is a number');
+console.log(numberX);
+numberX = 40;
+console.log('The new value of the X is 40');
```
Comment (eidosam) [inline Week1/Homework/js-exercises/Exercise3.js:6]: 👍

File: Week1/Homework/js-exercises/Exercise5.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+'use strict';
+const z = 7.25;
+console.log(z);
+const a = 7;
+console.log(a);
+if (a > z) {
+	console.log(a);
+} else {
+	console.log(z);
+}
```
Comment (eidosam) [inline Week1/Homework/js-exercises/Exercise5.js]: This shows the highest number between `a` and `z` but doesn't store it in a variable (for example `maxNumber`) as requested.
Hint; https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/max

File: Week1/Homework/js-exercises/Exercise6.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+let items;
```
Comment (eidosam) [inline Week1/Homework/js-exercises/Exercise6.js]: This is not an array!
Have a look at this document maybe can give you a hint https://www.w3schools.com/js/js_arrays.asp

File: Week1/Homework/js-exercises/Exercise6.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+let items;
+console.log("it's an empty array");
+console.log(items);
+let myFavoriteAnimals = [ 'Lion', 'Tiger', 'Elephant' ];
+console.log(myFavoriteAnimals);
+myFavoriteAnimals.push('Piglet');
```
Comment (eidosam) [inline Week1/Homework/js-exercises/Exercise6.js:7]: 👍

File: Week1/Homework/js-exercises/Exercise7.js [hunk 8]
```diff
@@ -0,0 +1,3 @@
+'use strict';
+const mySentence = 'Programming is so interesting!';
+console.log(mySentence.length);
```
Comment (eidosam) [inline Week1/Homework/js-exercises/Exercise7.js:3]: 👍

File: Week1/Homework/js-exercises/Exercise8.js [hunk 9]
```diff
@@ -0,0 +1,39 @@
+'use strict';
+let myName = 'mahmoud';
+let myLastName = 'badran';
+let FavoriteCars = [ 'BMW', 'Bugatti', 'Lamborghini' ];
+let FavoriteCities = [ 'Amsterdam', 'Damascus' ];
```
Comment (eidosam) [inline Week1/Homework/js-exercises/Exercise8.js]: Nice descriptive variable names! but remember to use camelCase for variables names.

File: Week1/Homework/js-exercises/Exercise8.js [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+	console.log('SAME TYPE');
+} else {
+	console.log('Not the same');
+}
+
+if (typeof FavoriteCities == typeof myLastName) {
+	console.log('SAME TYPE');
+} else {
+	console.log('Not the same');
+}
```
Comment (eidosam) [inline Week1/Homework/js-exercises/Exercise8.js:39]: Try to get used to the strict equality operator `===`, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comparison_Operators#Equality_operators

File: Week1/Homework/js-exercises/Exercise9.js [hunk 11]
```diff
@@ -0,0 +1,10 @@
+'use strict';
+let x = 7;
+x = x / 3;
+console.log(x);
+let y = 21;
+y = y / 4;
+console.log(y);
+let z = 13;
+z = z / 2;
+console.log(z);
```
Comment (eidosam) [inline Week1/Homework/js-exercises/Exercise9.js]: Can you write in a comment next to each calculation what will be the new value of each variable? 

> For each of these, write in comments what the answer is followed by how you came to that conclusion

File: Week1/Homework/js-exercises/Exercise10.js [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+
+console.log(items.length);
+
+console.log(colors.length);
+
+if (items.length == colors.length) {
+	console.log('same!');
+} else {
+	console.log('Two different sizes');
+}
```
Comment (eidosam) [inline Week1/Homework/js-exercises/Exercise10.js:22]: 👍

**Review by eidosam — CHANGES_REQUESTED** [review]
Well done, Mahmoud! There are little comments inline, please have a look and try to implement them, if you have any question, don't hesitate to ask!

File: Week1/Homework/js-exercises/Exercise10.js (hunk 12 above)
Comment (mahmoudbd) [reply Week1/Homework/js-exercises/Exercise10.js:22]: Hello Eidosam i have modified exercises 2,3,5,8,9  ,  could you please check it  after ((mahmoudbd wants to merge 2 commits into HackYourFuture:master from mahmoudbd:Mahmoud-JS-week1-HW))? Thanks

**Review by mahmoudbd — COMMENTED** [review]
Hello Eidosam i have modified exercises 2,3,5,8,9 , could you please check it after ((mahmoudbd wants to merge 2 commits into HackYourFuture:master from mahmoudbd:Mahmoud-JS-week1-HW))? Thanks

File: Week1/Homework/js-exercises/Exercise3.js [hunk 13]
```diff
@@ -0,0 +1,7 @@
+'use strict';
+let numberX;
+console.log('The value of X is undefined');
```
Comment (eidosam) [inline Week1/Homework/js-exercises/Exercise3.js:3]: 👍

File: Week1/Homework/js-exercises/Exercise6.js (hunk 6 above)
Comment (eidosam) [reply Week1/Homework/js-exercises/Exercise6.js]: To create an empty array; `const items = [];`

File: Week1/Homework/js-exercises/Exercise9.js [hunk 14]
```diff
@@ -0,0 +1,10 @@
+'use strict';
+let x = 7;
+x = x / 3;
+console.log(x); // expected output: 2.3333333333333335
+let y = 21;
+y = y / 4;
+console.log(y); //expected output: 5.25
+let z = 13;
+z = z / 2;
+console.log(z); //expected output: 6.5
```
Comment (eidosam) [inline Week1/Homework/js-exercises/Exercise9.js]: The operator you are requested to use is the remainder operator `%` not the division operator `/`
So for example the first case will be:
```
let x = 7;
x = x % 3;
console.log(x); // Expected output is 1, which is the remainder of dividing 7 by 3
```

File: Week1/Homework/js-exercises/Exercise5.js [hunk 15]
```diff
@@ -0,0 +1,6 @@
+'use strict';
+const z = 7.25;
+console.log(z);
+const a = 7;
+console.log(a);
+console.log(Math.max(z, a)); // expected output: 7.25
```
Comment (eidosam) [inline Week1/Homework/js-exercises/Exercise5.js]: The result need to be assigned to a variable, for example `const maxNumber = Math.max(z, a);`

**Review by eidosam — COMMENTED** [review]
Thanks Mahmoud for addressing the comments. There are little enhancements can be done, please see the new remarks.

**Comment by mahmoudbd** [comment]
Thank you very much for the help , could you please check it again?

**Review by eidosam — APPROVED** [review]
Thanks again Mahmoud for addressing the feedback, very well done!

**Pull request closed without merging**